biom/tests/test_table.py

```diff
@@ -1815,6 +1821,196 @@ def test_add_sample_metadata_two_entries(self):
         self.assertEqual(t._sample_metadata[2]['D'], ['A', 'C'])
         self.assertEqual(t._sample_metadata[3]['D'], ['A', 'D'])
 
+    def test_align_to_dataframe_samples(self):
+        table = Table(np.array([[0, 0, 1, 1],
+                                [2, 2, 4, 4],
+                                [5, 5, 3, 3],
+                                [0, 0, 0, 1]]).T,
+                      ['o1', 'o2', 'o3', 'o4'],
+                      ['s1', 's2', 's3', 's4'])
+        metadata = pd.DataFrame([['a', 'control'],
+                                 ['c', 'diseased'],
+                                 ['b', 'control']],
+                                index=['s1', 's3', 's2'],
+                                columns=['Barcode', 'Treatment'])
+        exp_table = Table(np.array([[0, 0, 1, 1],
+                                    [2, 2, 4, 4],
+                                    [5, 5, 3, 3]]).T,
+                          ['o1', 'o2', 'o3', 'o4'],
+                          ['s1', 's2', 's3'])
+        exp_metadata = pd.DataFrame([['a', 'control'],
+                                     ['b', 'control'],
+                                     ['c', 'diseased']],
+                                    index=['s1', 's2', 's3'],
+                                    columns=['Barcode', 'Treatment'])
+        res_table, res_metadata = table.align_to_dataframe(metadata)
+        pdt.assert_frame_equal(exp_metadata, res_metadata)
+        self.assertEqual(res_table.descriptive_equality(exp_table),
+                         'Tables appear equal')
+
+    def test_align_to_dataframe_observations(self):
+        table = Table(np.array([[0, 0, 1, 1],
+                                [2, 2, 4, 4],
+                                [5, 5, 3, 3],
+                                [0, 0, 0, 1]]),
+                      ['s1', 's2', 's3', 's4'],
+                      ['o1', 'o2', 'o3', 'o4'])
+        metadata = pd.DataFrame([['a', 'control'],
+                                 ['c', 'diseased'],
+                                 ['b', 'control']],
+                                index=['s1', 's3', 's2'],
```

Comment:
It would be nice for the naming to of identifiers to reflect that this is against observations as the current names imply it's against samples